### ISSUE TITLE
OneTimePassword as a pod dependency

### ISSUE THREAD

**Issue by darekxan** [mattrubin/OneTimePassword#1]
When I add OneTimePassword as a pod (checked out form my git fork, which has added *swift files as a source) I got following error:

```
~/OneTImePasswordPodTest/Pods/OneTimePassword/OneTimePassword/OneTimePassword-Bridging-Header.h:9:9: error: include of non-modular header inside framework module 'OneTimePassword.OneTimePassword_Bridging_Header'
#import <CommonCrypto/CommonHMAC.h>
        ^
<unknown>:0: error: could not build Objective-C module 'OneTimePassword'
```
I'm using CocoaPods 0.36.1

How to overcome this?

**Comment by mattrubin** [comment]
You may be able to specify the CommonCrypto dependency in the Podspec: http://guides.cocoapods.org/syntax/podspec.html#libraries

However, Xcode tends to throw warnings when non-modular code (like `libcommonCrypto.dylib`) is included in Framework modules (like `OneTimePassword.framework`). You may have to mess with this build setting (which Xcode refers to as "inherently unsafe"):

> CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES
> Enabling this setting allows non-modular includes to be used from within framework modules. This is inherently unsafe, as such headers might cause duplicate definitions when used by any client that imports both the framework and the non-modular includes themselves.

I've been exploring a number of workarounds for this problem, and while I can get the framework running just fine inside an application, I haven't found any solutions that don't throw compiler warnings of some kind. If you come up with a clean solution, I'd love to know what it is.

**Comment by darekxan** [comment]
I'm nowhere that advanced in iOS to figure this out. I've just started a few days ago. I'll keep this in mind, when I come across any usable info I'll report back.

**Comment by mattrubin** [comment]
I finally found a way to access CommonCrypto from a framework using CocoaPods. The solution involves [custom `modulemap` files and `SWIFT_INCLUDE_PATHS`][blog], and is implemented in [PR#59][pr].

[blog]: https://ind.ie/labs/blog/using-system-headers-in-swift/
[pr]: https://github.com/mattrubin/OneTimePassword/pull/59


This pod spec isn't yet available in the main CocoaPods spec repo, but you can now use the latest development version of OneTimePassword with this pod declaration:
```
pod 'OneTimePassword', :git => 'https://github.com/mattrubin/OneTimePassword'
```
